### PR TITLE
Pipelines v2 - Spotify

### PR DESCRIPTION
**Pull request by mre**
This is a complete rewrite of the pipeline architecture with some improvements:

* Fully typed
* Decoupling responsibilities to add unit tests
* Make code scale vertically and use job queue
* Make code more idiomatic

Endpoints can now be requested out of order, which makes the pipeline more flexible and parallelizable.

### CODE REVIEW

File: spotify/job/dates.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+        # Calculate the number of days between start and end date
+        self.days = (end - start).days
+
+    def __iter__(self):
+        """
+        Iterate over all days in the range
+        """
+        for i in range(self.days):
+            yield (self.start - dt.timedelta(days=i), self.end - dt.timedelta(days=i))
```
Comment (woolfg) [inline spotify/job/dates.py:23]: is this correct? you move the range and also return a range, shouldn't it be just one day?

File: spotify/job/test_dates.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+from job.dates import DateRange
+
+class TestDateRange(unittest.TestCase):
+    def setUp(self):
+        self.start = dt.datetime(2023, 1, 1)
+        self.end = dt.datetime(2023, 1, 5)
+        self.daterange = DateRange(self.start, self.end)
+
+    def test_iterator(self):
+        expected_output = [(dt.datetime(2023, 1, 1), dt.datetime(2023, 1, 5)),
```
Comment (woolfg) [inline spotify/job/test_dates.py:13]: why is the output a range? shouldn't it be just one day or the range of start===end?

**Review by woolfg — CHANGES_REQUESTED** [review]
architecture is nice, we went through it already in a call. nice!

the date issue looks a bit weird or do I misunderstand it?

File: spotify/job/dates.py (hunk 1 above)
Comment (mre) [reply spotify/job/dates.py:23]: it's not correct; will fix

**Review by woolfg — APPROVED** [review]

**Pull request merged**